### PR TITLE
Improving monthly report and combine report scripts

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Refactoring `combine-report`, `monthly-report`
* Introducing new `create-monthly-report` script

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/gvm-tools/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: scripts/create-consolidated-report.gmp.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+from gvmtools.helper import generate_uuid, error_and_exit
+
+HELP_TEXT = (
+    'This script creates a consolidated report and imports it to the GSM.'
+    ' Usable with gvm-script (gvm-tools)'
+)
+
+
+def parse_tags(tags: List):
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:33]: I would love to add always add a type hint for the content of the list. It is often required to read the whole function code to get the expected type.
```suggestion
def parse_tags(tags: List[str]):
```

File: scripts/create-consolidated-report.gmp.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        try:
+            UUID(tag, version=4)
+            filter_tags.append('tag_id="{}"'.format(tag))
+        except ValueError:
+            filter_tags.append('tag="{}"'.format(tag))
+
+    return filter_tags
+
+
+def parse_period(period: List):
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:52]:
```suggestion
def parse_period(period: List[str]):
```

File: scripts/create-consolidated-report.gmp.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    except ValueError as e:
+        error_and_exit('End date: {}'.format(e.args[0]))
+
+    if period_end < period_start:
+        error_and_exit('The start date seems to after the end date.')
+
+    return period_start, period_end
+
+
+def parse_args(args):  # pylint: disable=unused-argument
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:93]: Should get more type info including return type.

File: scripts/create-consolidated-report.gmp.py [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+        type=str,
+        dest='filter',
+        help='Filter the results by given filter(s).',
+    )
+
+    script_args, _ = parser.parse_known_args()
+    return script_args
+
+
+def generate_task_filter(period_start, period_end, tags: List):
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:150]: Should get more type info including return type.

File: scripts/create-consolidated-report.gmp.py [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+
+        tags_filter = ' or '.join(filter_parts)
+        task_filter += tags_filter
+    else:
+        task_filter += period_filter
+
+    return task_filter
+
+
+def get_last_reports_from_tasks(gmp, task_filter: str):
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:176]: Should get more type info including return type.

File: scripts/create-consolidated-report.gmp.py [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+    for report in tasks_xml.xpath('task/last_report/report/@id'):
+        reports.append(str(report))
+
+    # remove duplicates ... just in case
+    reports = list(dict.fromkeys(reports))
+
+    return reports
+
+
+def combine_reports(gmp, reports: List, filter_term: str):
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:197]: Should get more type info including return type.

File: scripts/create-consolidated-report.gmp.py [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+            ports_elem.append(port)
+        for result in current_report.xpath('report/results/result'):
+            results_elem.append(result)
+        for host in current_report.xpath('host'):
+            report_elem.append(host)
+
+    return combined_report
+
+
+def send_report(gmp, combined_report, period_start, period_end):
```
Comment (bjoernricks) [inline scripts/create-consolidated-report.gmp.py:238]: Should get more type info including return type.

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Looks good. Just please add more type hints for the new script.

File: scripts/create-consolidated-report.gmp.py (hunk 1 above)
Comment (y0urself) [reply scripts/create-consolidated-report.gmp.py:33]: Like i said: Improving `gvm-tools` :)

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Thanks a lot!

**Review by bjoernricks — APPROVED** [review]
Sorry did click on the wrong radio button ;-)

**Pull request merged**